tests/test_helpers.py

```diff
@@ -31,6 +31,48 @@ def test_escape_markdown(self):
 
         assert expected_str == helpers.escape_markdown(test_str)
 
+    def test_extract_urls_entities(self):
+        test_entities = [{
+            'length': 6, 'offset': 0, 'type': 'text_link',
+            'url': 'http://github.com/'
+        }, {
+            'length': 17, 'offset': 23, 'type': 'url'
+        }, {
+            'length': 14, 'offset': 43, 'type': 'text_link',
+            'url': 'http://google.com'
+        }]
+        test_text = 'Github can be found at http://github.com. Google is here.'
+        test_message = Message(message_id=1,
+                               from_user=None,
+                               date=None,
+                               chat=None,
+                               text=test_text,
+                               entities=[MessageEntity(**e) for e in test_entities])
+        result = helpers.extract_urls(test_message)
+
+        assert len(result) == 2
```

Comment:
I don;t get this. If there's three entities of type `url` and `text_link` in the message. Shouldn't `len(result) == 3`?

Reply:
extract_urls() strips trailing slashes and then dedupes, so the entity link `http://github.com/` and the inline link `http://github.com` are treated as the same.

(For what it's worth, in this case that's correct, but in cases where trailing slashes are anywhere else in the URL it's either unnecessary (`https://github.com/python-telegram-bot` versus `https://github.com/python-telegram-bot/`), meddlesome (any index page that uses relative links, and therefore must issue a redirect to restore the trailing slash), or actually broken (`http://www.example.com/?b64data=aGk/`).)